conans/client/graph/graph_builder.py

```diff
@@ -78,29 +79,34 @@ def extend_build_requires(self, graph, node, build_requires_refs, check_updates,
 
         conanfile = node.conanfile
         scope = conanfile.display_name
-        requires = [Requirement(ref) for ref in build_requires_refs]
-        for r in requires:
+
+        build_requires = []
+        for ref, context in build_requires_refs:
+            r = Requirement(ref)
             r.build_require = True
+            r.build_require_context = context
+            build_requires.append(r)
 
         if graph_lock:
             graph_lock.pre_lock_node(node)
-            graph_lock.lock_node(node, requires, build_requires=True)
+            # TODO: Add info about context?
+            graph_lock.lock_node(node, build_requires, build_requires=True)
```

Comment:
I think that the `build/host` information should be available in the graphlock, probably it requires a PR to this PR so we can merge this one first, but we can check how that one will look like.

Reply:
I prefer to go one thing at a time. At the moment this is self-contained, does it job, lets keep it focused. I don't think this will be necessary for 1.23, but let's see when the UI gets merged too, and we also have tests including lockfiles.